### PR TITLE
Added test coverage MFTF Entity name autocomplete and reference, fixed bugs

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description (*)
Hi @lbajsarowicz 
I've rechecked the functionality according to the issue description and fixed broken cases.

### Fixed Issues (if relevant)
1. magento/magento2-phpstorm-plugin#104: MFTF: Entity name autocomplete and reference

#### Create data ✅ 
image

#### Update data ✅ 
image

#### User input ✅ 
image

#### User input ✅
image 

#### Create data ✅ 
image

#### Update data ✅ 
image

#### Entity extends ✅
image

Also, I've covered these and many more cases with tests.
To be continued...

### CODE REVIEW

File: src/com/magento/idea/magento2plugin/reference/xml/XmlReferenceContributor.java [hunk 1]
```diff
@@ -157,22 +158,28 @@ public void registerReferenceProviders(@NotNull PsiReferenceRegistrar registrar)
...
 
         // <createData entity="SimpleProduct" />
+        // <updateData entity="SimpleProduct" />
         registrar.registerReferenceProvider(
-            XmlPatterns.xmlAttributeValue().withParent(XmlPatterns.xmlAttribute().withName(string().oneOf("entity", "value", "userInput", "url"))),
+            XmlPatterns.xmlAttributeValue().withParent(XmlPatterns.xmlAttribute()
+                    .withName(MftfActionGroup.ENTITY_ATTRIBUTE).
+                    withParent(XmlPatterns.xmlTag().withName(
+                      string().oneOf(MftfActionGroup.CREATE_DATA_TAG, MftfActionGroup.UPDATE_DATA_TAG)
+                    ))),
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/reference/xml/XmlReferenceContributor.java:180]:
```suggestion
            XmlPatterns.xmlAttributeValue().withParent(XmlPatterns.xmlAttribute()
                .withName(MftfActionGroup.ENTITY_ATTRIBUTE)
                .withParent(XmlPatterns.xmlTag().withName(
                    string().oneOf(MftfActionGroup.CREATE_DATA_TAG, MftfActionGroup.UPDATE_DATA_TAG)
            ))),
```

File: tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+package com.magento.idea.magento2plugin.completion.xml;
+
+public class MftfEntityNameCompletionRegistrarTest extends CompletionXmlFixtureTestCase {
+
+    private static final String[] lookupStringsEntities = new String[]{
+                "TestAdminMenuCatalog",
+                "TestAdminMenuCatalog.pageTitle",
+                "TestDefaultAttributeSet",
+                "TestDefaultAttributeSet.attribute_set_id"
+        };
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java:14]:
```suggestion
    private static final String[] lookupStringsEntities = new String[] {
        "TestAdminMenuCatalog",
        "TestAdminMenuCatalog.pageTitle",
        "TestDefaultAttributeSet",
        "TestDefaultAttributeSet.attribute_set_id"
      };
```

File: tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+    public void testCreateDataActionGroupMustProvideCompletion () {
+        String filePath = this.getFixturePath(
+                "TestActionGroup.xml"
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java:18]:
```suggestion
          "TestActionGroup.xml"
```

File: tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+public class MftfEntityNameCompletionRegistrarTest extends CompletionXmlFixtureTestCase {
+
+    private static final String[] lookupStringsEntities = new String[]{
+                "TestAdminMenuCatalog",
+                "TestAdminMenuCatalog.pageTitle",
+                "TestDefaultAttributeSet",
+                "TestDefaultAttributeSet.attribute_set_id"
+        };
+
+    public void testCreateDataActionGroupMustProvideCompletion () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java:16]:
```suggestion
    public void testCreateDataActionGroupMustProvideCompletion() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 5]
```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+package com.magento.idea.magento2plugin.reference.xml;
+
+public class MftfEntityNameReferenceRegistrarTest extends ReferenceXmlFixtureTestCase {
+
+    public void testCreateDataActionGroupMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:9]: 🤔 Is there any specific method declaration in Java?
```suggestion
    public void testCreateDataActionGroupMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java (hunk 3 above)
Comment (VitaliyBoyko) [reply tests/com/magento/idea/magento2plugin/completion/xml/MftfEntityNameCompletionRegistrarTest.java]: ✅

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java (hunk 5 above)
Comment (VitaliyBoyko) [reply tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:9]: No, it's typo

Comment (VitaliyBoyko) [reply tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:9]: :)

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+public class MftfEntityNameReferenceRegistrarTest extends ReferenceXmlFixtureTestCase {
+
+    public void testCreateDataActionGroupMustHaveReference() {
+        String filePath = this.getFixturePath("TestActionGroup.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testCreateDataInTestMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:16]:
```suggestion
    public void testCreateDataInTestMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testUpdateDataActionGroupMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:23]:
```suggestion
    public void testUpdateDataActionGroupMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestActionGroup.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testUpdateDataInTestMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:30]:
```suggestion
    public void testUpdateDataInTestMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testUserInputInActionGroupMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:37]:
```suggestion
    public void testUserInputInActionGroupMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestActionGroup.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testUserInputInTestMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:44]:
```suggestion
    public void testUserInputInTestMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testEntityExtendsInDataMustHaveReference () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:51]:
```suggestion
    public void testEntityExtendsInDataMustHaveReference() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 12]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestData.xml");
+        myFixture.configureByFile(filePath);
+
+        assertHasReferenceToXmlAttributeValue("TestAdminMenuCatalog");
+    }
+
+    public void testCreateDataActionGroupPluginDisabled () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:58]:
```suggestion
    public void testCreateDataActionGroupPluginDisabled() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestActionGroup.xml");
+        myFixture.configureByFile(filePath);
+        disablePluginAndReindex();
+
+        assertEmptyReference();
+    }
+
+    public void testCreateDataActionGroupMftfSupportDisabled () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:66]:
```suggestion
    public void testCreateDataActionGroupMftfSupportDisabled() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestActionGroup.xml");
+        myFixture.configureByFile(filePath);
+        disableMftfSupportAndReindex();
+
+        assertEmptyReference();
+    }
+
+    public void testCreateDataInTestWithSectionMustBeEmpty () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:74]:
```suggestion
    public void testCreateDataInTestWithSectionMustBeEmpty() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertEmptyReference();
+    }
+
+    public void testUpdateDataInTestWithSectionMustBeEmpty () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:81]:
```suggestion
    public void testUpdateDataInTestWithSectionMustBeEmpty() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertEmptyReference();
+    }
+
+    public void testUserInputInTestMustBeEmpty () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:88]:
```suggestion
    public void testUserInputInTestMustBeEmpty() {
```

File: tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+        String filePath = this.getFixturePath("TestMftfTest.xml");
+        myFixture.configureByFile(filePath);
+
+        assertEmptyReference();
+    }
+
+    public void testEntityExtendsInDataMustBeEmpty () {
```
Comment (eduard13) [inline tests/com/magento/idea/magento2plugin/reference/xml/MftfEntityNameReferenceRegistrarTest.java:95]:
```suggestion
    public void testEntityExtendsInDataMustBeEmpty() {
```

**Pull request merged**